File: frontend/admin/src/api/qa.ts
export interface QAPair {
  id: string
  question: string
  answer: string
  keywords: string[]
  category: string | null
  is_active: boolean
  created_at: string
  updated_at: string
}

export async function listQA(params?: {
  search?: string
  category?: string
  include_inactive?: boolean
}): Promise<QAPair[]> {
  const q = new URLSearchParams()
  if (params?.search) q.set('search', params.search)
  if (params?.category) q.set('category', params.category)
  if (params?.include_inactive) q.set('include_inactive', 'true')
  const r = await fetch(`/api/v1/qa${q.size ? `?${q}` : ''}`)
  if (!r.ok) throw new Error(`HTTP ${r.status}`)
  return r.json()
}

export async function createQA(data: Omit<QAPair, 'id' | 'created_at' | 'updated_at'>): Promise<QAPair> {
  const r = await fetch('/api/v1/qa', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
  if (!r.ok) throw new Error(`HTTP ${r.status}`)
  return r.json()
}

export async function updateQA(id: string, data: Partial<QAPair>): Promise<QAPair> {
  const r = await fetch(`/api/v1/qa/${id}`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
  if (!r.ok) throw new Error(`HTTP ${r.status}`)
  return r.json()
}

export async function deleteQA(id: string): Promise<void> {
  const r = await fetch(`/api/v1/qa/${id}`, { method: 'DELETE' })
  if (!r.ok) throw new Error(`HTTP ${r.status}`)
}

export async function batchImportQA(
  items: Array<{ question: string; answer: string; keywords?: string[]; category?: string }>
): Promise<{ imported: number }> {
  const r = await fetch('/api/v1/qa/batch', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ items }),
  })
  if (!r.ok) throw new Error(`HTTP ${r.status}`)
  return r.json()
}

File: backend/app/api/v1/qa.py
from __future__ import annotations

import uuid

from fastapi import APIRouter, Depends, Query
from pydantic import BaseModel
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.schemas.qa_pair import QAPairCreate, QAPairResponse, QAPairUpdate
from app.services import qa_service

router = APIRouter(prefix="/qa", tags=["qa"])


@router.get("", response_model=list[QAPairResponse])
async def list_qa(
    search: str | None = Query(default=None),
    category: str | None = Query(default=None),
    include_inactive: bool = Query(default=False),
    limit: int = Query(default=100, ge=1, le=500),
    db: AsyncSession = Depends(get_db),
) -> list[QAPairResponse]:
    pairs = await qa_service.list_qa(
        db, search=search, category=category,
        include_inactive=include_inactive, limit=limit,
    )
    return [QAPairResponse.model_validate(p) for p in pairs]


@router.post("", response_model=QAPairResponse, status_code=201)
async def create_qa(
    body: QAPairCreate,
    db: AsyncSession = Depends(get_db),
) -> QAPairResponse:
    pair = await qa_service.create_qa(
        db, question=body.question, answer=body.answer,
        keywords=body.keywords, category=body.category,
    )
    return QAPairResponse.model_validate(pair)


@router.put("/{qa_id}", response_model=QAPairResponse)
async def update_qa(
    qa_id: uuid.UUID,
    body: QAPairUpdate,
    db: AsyncSession = Depends(get_db),
) -> QAPairResponse:
    updates = body.model_dump(exclude_unset=True)
    pair = await qa_service.update_qa(db, qa_id, **updates)
    return QAPairResponse.model_validate(pair)


@router.delete("/{qa_id}", status_code=200)
async def delete_qa(
    qa_id: uuid.UUID,
    db: AsyncSession = Depends(get_db),
) -> None:
    await qa_service.delete_qa(db, qa_id)


class BatchImportRequest(BaseModel):
    items: list[QAPairCreate]


class BatchImportResponse(BaseModel):
    imported: int


@router.post("/batch", response_model=BatchImportResponse)
async def batch_import(
    body: BatchImportRequest,
    db: AsyncSession = Depends(get_db),
) -> BatchImportResponse:
    items = [i.model_dump() for i in body.items]
    count = await qa_service.batch_import(db, items)
    return BatchImportResponse(imported=count)
